python/tank/bootstrap/configuration.py

```diff
@@ -92,10 +92,20 @@ def get_tk_instance(self, sg_user):
         # by creating a pipeline configuration object directly
         # and pass that into the factory method.
 
-        pc = pipelineconfig.PipelineConfiguration(path, self.descriptor)
+        if self._does_pipelineconfigution_supports_descriptor(pipelineconfig.PipelineConfiguration):
+            pc = pipelineconfig.PipelineConfiguration(path, self.descriptor)
```

Comment:
So the reason to supply the `descriptor` via the constructor is so that it has a handle on it and can be accessed later on for whatever reason (that right?). Would it be possible, or make sense, to use a setter on the pipeline configuration post init instead just to avoid changing the API?